open-apparel-registry/open-apparel-registry
# Fix embedded map styling

## Overview

The embedded map custom styles weren't applied to all relevant elements when creating the new layout. Applies the custom color/font where applicable, and also tweaks a few small UI style issues. 

Connects Client 115

## Demo

<img width="1405" alt="Screen Shot 2022-11-03 at 12 09 11 PM" src="https://user-images.githubusercontent.com/21046714/199788907-e27f84c5-e2fe-4cd1-bbf3-994a92ba1318.png">

<img width="492" alt="Screen Shot 2022-11-03 at 12 34 24 PM" src="https://user-images.githubusercontent.com/21046714/199788924-bc60b8d7-10aa-4645-9fe9-49f299d3233d.png">

With Contrast Text:
<img width="1408" alt="Screen Shot 2022-11-03 at 2 29 09 PM" src="https://user-images.githubusercontent.com/21046714/199805228-d9f5cb79-ed03-4680-b1f3-4e3df306e516.png">
<img width="497" alt="Screen Shot 2022-11-03 at 2 29 22 PM" src="https://user-images.githubusercontent.com/21046714/199805239-8054cfeb-a83d-4d7e-8d23-832df5eaa1e5.png">

## Notes

I preferred using the `theme` over pulling in props for custom color/font where possible, to simplify style adjustments later if needed (since the `theme` object is in one place, vs many mapStateToProps). 

Currently, we have three main colors in the app: purple, pink, and yellow. The purple was previously set to be the primary color, so I've left that, and added the pink as secondary, and the yellow as 'action' (since it mostly appears in actionable items, like the search button). The custom type ('action') doesn't have the same power as the built-in primary/secondary color schemes, but is still useful as a variable that can be accessed via theming throughout the app. 

We only allow users to select one color in embedded map config, so in embed mode I've set all the colors to the config color.  However, if we decide to allow users to set multiple colors in the future, it should be fairly simple to apply that via the theme object. 

## Testing Instructions

* Run `./scripts/server`
* View [the homepage](http://localhost:6543/) at full width and mobile. Select two contributors, and view the tooltip for the contributors filter. Note font, color, and alignment throughout.
* View [the facilities page](http://localhost:6543/facilities) at full with and mobile. On mobile, open the filters drawer. Select two contributors, and view the tooltip for the contributors filter. Note font, color, and alignment throughout.
* Go to user embed settings. Note that text should read, 'OS ID' and not 'OS Hub ID'. 
* Set up an embedded map with a very noticeable color and font. Set a field to be searchable, and update the search text. 
* View [the embedded facilities page](http://localhost:6543/facilities?contributors=2&embed=1) at full width and mobile. On mobile, open the filters drawer. Note the alignment of the 'powered by' button. Note font, color, and alignment throughout.
* Test the embedded map again with a user with lowest embed permissions, and the map color set to 'black'.

## Checklist
